### PR TITLE
Replace date validation of recurring transactions ...

### LINKED ISSUE #16: Recurring transactions fail to submit if day is lower than 10 or higher than 28
#### Actual behaviour
When creating a transaction with the "This is a recurring spending" checkbox checked, submitting the transaction will fail if the day is `01` to `09` or `>28`. 
`validator.integer` will be displayed under input field for "day". 

#### Expected behaviour: 
When creating a transaction with a day lower then 10 or greater than 28, submitting will persist the recurring transaction as a row in the database. 

#### Steps to reproduce: 

* visit `/transactions/create` 
* enter spending transaction
* use date with single digit day or greater than 28
* check "This is a recurring spending" checkbox
* fill other required inputs
* submit 

#### Cause

From what I can see, the validation checks for a date that is `between:1,28`. 
This causes two possible scenarios for the validation to fail: 

1. for days from 1 to 9, the date picker will cause the component to send `01` to `09` as input, causing the validator to reject due to incompatibility of `01` and `1`, for example. 
2. for days greater than 28 the validator will reject due to the number being higher than the allowed maximum.

#### Contribution?
I initially wanted to write something myself, but found this gem instead! 
This is great work already! 
I'd be happy to contribute a fix, in case you'd be interested :)

### PR DESCRIPTION
**Pull request by oliverjkb**
... to make transactions possible at every day of month

As per the issue I submitted, recurring transactions could not be entered because of date incompatibilities. This fixes #16

### CODE REVIEW

**Comment by range-of-motion** [comment]
@oliverjkb sorry for the late response. I've been busy and this project has been on the down low for a while. 

Anyway. 

What's supposed to happen when I enter a transaction for the 31st of each month, on months that only go to let's say 30 days?

**Comment by oliverjkb** [comment]
No worries, we are only humans in the end.. ;) 

I must admit, I did not think about that too much.  
It would, however, fit to process these recurrings like a bank would do, and check if the last day of the month is reached and if there are recurrings with a "higher" day to be processed.  

That way all recurrings that would occur on a day higher than 28 would be processed on the last day of a month in february, all recurrings with a day of 31 on the 30st in months falling short of one day. 

Thinking about this, I also recognize that recurrings are not being dispatched in `RecurringController`. 
Also, a possible job would perish after running once, but we would need them to be despatched every day to make sure they are repeated until `ends_on` date is reached.  
Is this accounted for somewhere else?

**Comment by range-of-motion** [comment]
> It would, however, fit to process these recurrings like a bank would do, and check if the last day of the month is reached and if there are recurrings with a "higher" day to be processed.
> 
> That way all recurrings that would occur on a day higher than 28 would be processed on the last day of a month in february, all recurrings with a day of 31 on the 30st in months falling short of one day.

Good one. Hadn't thought about it like that. Does your PR feature that?

Not sure what you mean with your last paragraph?

**Comment by oliverjkb** [comment]
> Does your PR feature that?  

Not yet, but can definitly add that feature. :smiley: 

> Not sure what you mean with your last paragraph? 

I tried to make it more clear in issue #20 - fixing it should only be a few lines. I can put that in as well.

**Comment by range-of-motion** [comment]
Sure, let me know when you're ready.

**Comment by oliverjkb** [comment]
We should be ready now.

File: app/Jobs/ProcessRecurrings.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
 use Illuminate\Foundation\Bus\Dispatchable;
 use App\Recurring;
 use App\Spending;
+use Illuminate\Support\Facades\Log;
```
Comment (range-of-motion) [inline app/Jobs/ProcessRecurrings.php]: ..?

**Comment by range-of-motion** [comment]
Good job on the PR. Apart from getting rid of the `Log` usage I'd say we're solid.

I'd also like to make a unit test for recurrings, but we can do that later.

File: app/Jobs/ProcessRecurrings.php (hunk 1 above)
Comment (oliverjkb) [reply app/Jobs/ProcessRecurrings.php]: Jip. That was just used for logging during debugging. Just fixed that. ;)

**Comment by range-of-motion** [comment]
Do you mind squashing your commits? I like to keep a somewhat clean history and for example 5c72b7c or 96a132b just pollute it.

**Comment by oliverjkb** [comment]
Nope, no worries at all. :+1:

**Pull request merged**